projects/common/src/time/user-specified-time-range.service.ts

```diff
@@ -54,23 +34,37 @@ export class UserSpecifiedTimeRangeService {
         }
 
         return this.timeRangeService.timeRangeFromUrlString(timeRanges[path]);
-      }),
-      share()
+      })
     );
   }
 
   public setUserSpecifiedTimeRangeForPage(path: string, value: TimeRange): void {
-    const pageTimeMap: PageTimeRangeMap = this.pageTimeRangesSubject$.getValue();
-
-    const newMap: PageTimeRangeMap = { ...pageTimeMap, [path]: value.toUrlString() };
+    const newMap: PageTimeRangeStringDictionary = { ...(this.currentTimeRangeMap ?? {}), [path]: value.toUrlString() };
```

Comment:
Still problematic, actually even more so- if set is called before current is assigned, this would overwrite the saved values. Current is assigned in tap, which means it only is assigned as an async side effect of calling get (which is quite confusing!).

Instead, as I mentioned above, I would just basically change this to call something like
```ts
this.getTimeRangeStringDictionary()
  .pipe(
    take(1), // If we don't want to watch the value on read, should actually do this in the get call too
    map(current => ({...current, [path], value.toUrlString()})
  )
  .subscribe(merged => this.preferenceService.set(...));